addons/web/static/src/views/view_dialogs/export_data_dialog.js

```diff
@@ -23,18 +23,28 @@ DeleteExportListDialog.template = "web.DeleteExportListDialog";
 class ExportDataItem extends Component {
     setup() {
         this.state = useState({
-            isExpanded: this.subFields.length > 0 && this.props.isExpanded,
+            subfields: [],
+        });
+        onWillStart(() => {
+            if (this.props.isExpanded) {
+                // automatically expand the item when subfields are already loaded
+                // and display subfields that match the search string
+                this.loadExpandedContent(this.props.field.id, false);
+            }
         });
     }
 
-    get subFields() {
-        return this.props.getSubFields(this.props.field.id);
-    }
-
-    async loadExpandedContent(id) {
+    async loadExpandedContent(id, isUserToggle) {
         if (this.props.isFieldExpandable(id)) {
-            await this.props.onToggleExpandField(id);
-            this.state.isExpanded = !this.state.isExpanded;
+            if (!this.state.subfields.length) {
+                const subfields = await this.props.onToggleExpandField(id, !isUserToggle);
+                if (subfields) {
+                    return (this.state.subfields = isUserToggle
+                        ? subfields
+                        : this.props.filterSubfields(subfields));
+                }
+            }
+            this.state.subfields = [];
```

Comment:
not sure to understand why you reset it if there were subfields

Reply:
It is done here to handle when the toggle is closed, and when there are no subfields.
I may not be that clear, but there is a return statement line 42 when they were subfields

Reply:
Yep I saw the return line 42. I'm confused with the if line 39. So this function is called when the field is toggled (off)? That's a bit weird considering the name of the function